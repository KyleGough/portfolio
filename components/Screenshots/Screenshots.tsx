import { ImageModal } from '@components/ImageModal';
import { Section } from '@components/Section';
import { StaticImageData } from 'next/image';
import React from 'react';

interface ScreenshotProps {
  images: {
    imageData: StaticImageData;
    alt: string;
  }[];
}

export const Screenshots: React.FC<ScreenshotProps> = ({ images }) => (
  <Section>
    <h2 className="text-3xl mb-12">Screenshots</h2>
    <div className="grid grid-cols-12 gap-4">
      {images.map((image, i) => {
        return <ImageModal key={i} src={image.imageData.src} alt={image.alt} />;
      })}
    </div>
  </Section>
);
