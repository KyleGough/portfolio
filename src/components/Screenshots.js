import React from 'react';
import Section from '../components/Section';
import ImageModal from '../components/ImageModal';

export default function Screenshots(props) {
    return (
        <Section>
            <h2 className='text-3xl mb-12'>Screenshots</h2>
            <div className='grid grid-cols-12 gap-4'>
                {props.images.map((image, i) => {
                    return (
                        <ImageModal key={i} src={image.src} caption={image.caption} />    
                    )
                })}
            </div>
        </Section>
    );
}