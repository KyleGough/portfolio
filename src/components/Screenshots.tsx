import Section from '../components/Section';
import ImageModal from '../components/ImageModal';

type ScreenshotProps = {
    images: {
        src: string,
        caption: string
    }[]
};

export default function Screenshots(props: ScreenshotProps) {
    return (
        <Section>
            <h2 className='text-3xl mb-12'>Screenshots</h2>
            <div className='grid grid-cols-12 gap-4'>
                {props.images.map((image, i) => {
                    return (
                        <ImageModal key={i} src={image.src} caption={image.caption} />    
                    )
                })}
            </div>
        </Section>
    );
}
