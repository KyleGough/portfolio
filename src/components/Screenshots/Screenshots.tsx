import React from 'react';
import { Section } from '../Section';
import { ImageModal } from '../ImageModal/ImageModal';

interface ScreenshotProps {
  images: {
    src: string;
    caption: string;
  }[];
}

export const Screenshots: React.FC<ScreenshotProps> = ({ images }) => (
  <Section>
    <h2 className="text-3xl mb-12">Screenshots</h2>
    <div className="grid grid-cols-12 gap-4">
      {images.map((image, i) => {
        return <ImageModal key={i} src={image.src} caption={image.caption} />;
      })}
    </div>
  </Section>
);
