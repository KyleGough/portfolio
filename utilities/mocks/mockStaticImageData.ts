const mockStaticImageData = {
  src: '/#image',
  width: 100,
  height: 100,
  blurDataURL: '/#blur',
};

export default mockStaticImageData;
