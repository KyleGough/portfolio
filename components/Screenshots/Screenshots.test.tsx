import mockStaticImageData from '@mocks/mockStaticImageData';
import { render, screen } from '@testing-library/react';
import React from 'react';

import { Screenshots } from './Screenshots';

describe('Screenshots component', () => {
  it('renders', async () => {
    const { container } = render(
      <Screenshots
        images={[
          { imageData: mockStaticImageData, alt: 'Image A' },
          { imageData: mockStaticImageData, alt: 'Image B' },
        ]}
      />
    );

    expect(screen.getByText('Image A')).toBeVisible();
    expect(screen.getByText('Image B')).toBeVisible();
    expect(container).toMatchSnapshot();
  });
});
