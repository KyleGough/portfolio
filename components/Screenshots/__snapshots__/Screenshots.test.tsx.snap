// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screenshots component renders 1`] = `
<div>
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <h2
      class="text-3xl mb-12"
    >
      Screenshots
    </h2>
    <div
      class="grid grid-cols-12 gap-4"
    >
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-6"
      >
        <img
          alt="Image A"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/#"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Image A
        </figcaption>
      </figure>
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-6"
      >
        <img
          alt="Image B"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/#"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Image B
        </figcaption>
      </figure>
    </div>
  </div>
</div>
`;
